documentation/Software Requirements Specification.md

```diff
@@ -0,0 +1,333 @@
+<br>Pic. 2.2 Add client.
+
+When adding a client, the following details are entered:
+* Last name – client’s last name; 
+* First name – client’s first name; 
+* Phone - the client's phone number;
+* Registration date – client’s date registration; 
+* Passport number – the unique number of the client's passport;
+
+Constraints for data validation:
+* Last name – maximum length of 45 characters; 
+* First name –  maximum length of 45 characters; 
+* Phone –  maximum length of 30 characters; 
+* Registration date – client’s registration date in format dd/mm/yyyy
+* Passport number - maximum length of 45 characters;
+
+### 2.3 Edit client
+***Main scenario:***
+* User clicks the “Edit” button in the clients list view mode; 
+* Application displays form to enter client data; 
+* User enters client’s data and presses “Save” button; 
+* If any data is entered incorrectly, incorrect data messages are displayed; 
+* If entered data is valid, then edited data is added to database; 
+* If error occurs, then error message is displaying; 
+* If client’s record is successfully edited, then list of clients with added records is displaying. 
+
+***Cancel operation scenario:***
+* User clicks the “Edit” button in the clients list view mode; 
+* Application displays form to enter client data; 
+* User enters client data and presses “Cancel” button; 
+* Data don’t save in data base, then list of clients records is displaying to user. 
+* If the user selects the menu item "Orders”, ”Clients” or "Cars", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+
+![](edit_client.PNG)
+<br>Pic. 2.3 Edit client.
+### 2.4 Removing client
+***Main scenario:***
+* The user, while in the list of clients mode, presses the "Delete" button in the selected client line; 
+* Application displays confirmation dialog “Please confirm delete client?”; 
+* The user confirms the removal of the client; 
+* Record is deleted from database; 
+* If error occurs, then error message displays; 
+* When a client is deleted, a confirmation window appears, and the row in the table of this client turns red. The data in the other tables does not change;
+
+![](delete_client.PNG)
+<br>Pic. 2.4 Delete client dialog.
+
+***Cancel operation scenario:***
+* User is in display mode of clients list and press “Delete” button; 
+* Application displays confirmation dialog “Please confirm delete client?”;
+* User press “Cancel” button; 
+* List of clients without changes is displaying.
+
+
+
+## 3.  Cars
+
+### 3.1 Display list of cars
+This mode is intended for viewing and editing the cars list.
+***Main scenario:***
+* User selects item “Cars”; 
+* Application displays list of cars.
+
+
+![](Cars.PNG)
+<br>Pic. 3.1 View the cars list.
+
+* Brand  –  brand car; 
+* Model - car model;
+* Price - the price of the car;
+* VIN - Vehicle identification number;
+* Year of release - the year of the car 's release
+* Color - the color of the car
```

Comment:
Why you have not added the Count field of field that indicate whether the car is available to order?